### DISCUSSION TITLE
Make new releases periodically

### DISCUSSION THREAD

**Discussion by bansan85** [mosra/magnum#615] [General]
The latest official release for Corrade/Magnum is 2020.06.

Before, there were one release each 6 months. This was great.

If I want a new version, I don't know which commit for Corrade/Magnum should I take. Both master ? If the latest version doesn't fit me, it's also impossible to know which Corrade commit is compatible with Magnum commit and which commit is stable.

I know there is `next` branch. But there is no "merge commit" in master. So it's not possible to know in master if a commit is a final commit in a previous next branch or a WIP commit.

**Comment by mosra** [comment]
Hi, sorry, unfortunately I don't have the bandwidth to make releases often, even though I'm _really trying to_. The ideal case would be releases every 6 months as you say, yes. But current situation is that there's many features being in-progress due to extremely high demands from multiple projects, and I don't have any time left to do the remaining work that would make them release-ready. Such as demos, tutorials, examples and introductionary posts on the web. And tagging a release with nothing to present / advertise would be a wasted opportunity. Further details as well as a list of yet-unfinished TODOs is at https://github.com/mosra/magnum/issues/453.

What I can guarantee instead is that _all_ commits are always stable, so relying on `master` for all repos is fine. When you want to update you simply pull latest `master` of all repos again. That's what most users do, and I'm putting in a lot of effort to ensure `master` is always stable, documented and fully covered with tests. Commits go to `master` only once they compile and pass tests on all supported platforms.

The `next` branch is just a temporary branch [that I run tests on](https://magnum.graphics/build-status/) right before pushing to `master`. Without any merge commit because those are just noise -- the individual commits make sense on their own, and there's never any partial or WIP commit.

Hope this helps!

**Reply by bansan85** [reply]
Thanks for taking time to answer and explanation. I understand releasing takes time. Good luck with your huge todo list and thanks for your work.
